Detectors/TRD/raw/include/TRDRaw/DigitsParser.h

```diff
@@ -0,0 +1,49 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   EpnRawReaderTask.h
+/// @author Sean Murray
+/// @brief  TRD cru output data to tracklet task
+
+#ifndef O2_TRD_DIGITSPARSER
+#define O2_TRD_DIGITSPARSER
+
+#include "Framework/Task.h"
+#include "Framework/DataProcessorSpec.h"
+#include "TRDRaw/CruRawReader.h"
+#include "TRDRaw/CompressedRawReader.h"
+#include <fstream>
+
+using namespace o2::framework;
+
+namespace o2
+{
+namespace trd
+{
+class Digit;
+
+class DigitsParser 
+{
+ public:
+  DigitsParser() = default;
+  ~DigitsParser() = default;
+ uint32_t Parse(std::vector<uint64_t> *data){setData(data); return Parse();};
```

Comment:
It seems this would be cleaner with a span or begin/end iterators. But maybe that's what you commented on during the meeting.

Reply:
data holds a complete cruhalfchamber data block, a span would be cleaner, and probably preferable.